commons/src/main/java/org/infinispan/commons/api/AsyncCache.java

```diff
@@ -265,20 +269,37 @@
    CompletableFuture<Boolean> replaceAsync(K key, V oldValue, V newValue, long lifespan, TimeUnit lifespanUnit, long maxIdle, TimeUnit maxIdleUnit);
 
    /**
-    * Asynchronous version of {@link #get(Object)} that allows user code to
+    * Asynchronous version of {@link BasicCache#get(Object)} that allows user code to
     * retrieve the value associated with a key at a later stage, hence allowing
     * multiple parallel get requests to be sent. Normally, when this method
     * detects that the value is likely to be retrieved from from a remote
     * entity, it will span a different thread in order to allow the
     * asynchronous get call to return immediately. If the call will definitely
     * resolve locally, for example when the cache is configured with LOCAL mode
     * and no stores are configured, the get asynchronous call will act
-    * sequentially and will have no different to {@link #get(Object)}.
+    * sequentially and will have no different to {@link BasicCache#get(Object)}.
     *
     * @param key key to retrieve
     * @return a future that can be used to retrieve value associated with the
     * key when this is available. The actual value returned by the future
-    * follows the same rules as {@link #get(Object)}
+    * follows the same rules as {@link BasicCache#get(Object)}
     */
    CompletableFuture<V> getAsync(K key);
+
+   /**
+    * Asynchronous version of {@link BasicCache#containsKey(Object)}
+    * @param key key to retrieve
+    * @return future containing true if the mapping exists.
+    */
+   default CompletableFuture<Boolean> containsKeyAsync(K key) {
+      return getAsync(key).thenApply(Objects::nonNull);
+   }
+
+   default CompletableFuture<Map<K, V>> getAllAsync(Set<?> keys) {
```

Comment:
I was just wondering about this yesterday. We had a `putAll` but there was no `getAll` 👍